lib/Runtime/Library/JavascriptExceptionMetadata.cpp

```diff
@@ -0,0 +1,144 @@
+
+        var = Js::JavascriptOperators::ToNumber(
+            Js::JavascriptOperators::OP_GetProperty(exception, Js::PropertyIds::column, scriptContext),
+            scriptContext);
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::column, var, scriptContext);
+
+        var = Js::JavascriptOperators::ToNumber(
+            Js::JavascriptOperators::OP_GetProperty(exception, Js::PropertyIds::length, scriptContext),
+            scriptContext);
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::length, var, scriptContext);
+
+        var = Js::JavascriptConversion::ToString(
+            Js::JavascriptOperators::OP_GetProperty(exception, Js::PropertyIds::source, scriptContext),
+            scriptContext);
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::source, var, scriptContext);
+
+        var = Js::JavascriptConversion::ToString(
+            Js::JavascriptOperators::OP_GetProperty(exception, Js::PropertyIds::url, scriptContext),
+            scriptContext);
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::url, var, scriptContext);
+    }
+
+    bool JavascriptExceptionMetadata::PopulateMetadataFromException(Var metadata, JavascriptExceptionObject * recordedException, ScriptContext * scriptContext) {
+        uint32 offset = recordedException->GetByteCodeOffset();
+        FunctionBody * functionBody = recordedException->GetFunctionBody();
+        ULONG line;
+        LONG column;
+        if (functionBody->GetUtf8SourceInfo()->GetIsLibraryCode() ||
+            !functionBody->GetLineCharOffset(offset, &line, &column)) {
+            line = 0;
+            column = 0;
+        }
+
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::line,
+            Js::JavascriptNumber::New(line, scriptContext), scriptContext);
+
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::column,
+            Js::JavascriptNumber::New(column, scriptContext), scriptContext);
+
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::length,
+            Js::JavascriptNumber::New(0, scriptContext), scriptContext);
+
+        Js::JavascriptOperators::OP_SetProperty(metadata, Js::PropertyIds::url,
+            Js::JavascriptString::NewCopySz(functionBody->GetSourceContextInfo()->url, scriptContext), scriptContext);
+
+        Js::Utf8SourceInfo* sourceInfo = functionBody->GetUtf8SourceInfo();
+        sourceInfo->EnsureLineOffsetCache();
+        JsUtil::LineOffsetCache<Recycler> *cache = sourceInfo->GetLineOffsetCache();
+
+        LPCUTF8 functionSource = sourceInfo->GetSource(_u("Jsrt::JsExperimentalGetAndClearExceptionWithMetadata"));
+
+        charcount_t startByteOffset;
+        charcount_t endByteOffset;
+        charcount_t startCharOffset;
+        charcount_t endCharOffset;
+
+        if (line >= cache->GetLineCount())
+        {
+            return false;
```

Comment:
if we return false here, all the setproperty we have done above is wasted, isn't it? make sense to move this to top of the function.